src/util.ts

```diff
@@ -419,9 +419,7 @@ export class Util {
       try {
         parsedHttpRespBody.body = JSON.parse(body);
       } catch (err) {
-        parsedHttpRespBody.err = new ApiError(
-          `Cannot parse response as JSON: ${body}`
-        );
+        parsedHttpRespBody.body = body;
```

Comment:
@bcoe this is one of those "technically a breaking change" issues where we're changing the type of the error being returned.  I uh, don't know what the right answer is here.

Reply:
@jkwlui could we potentially look at response headers, and continue to throw if the response is `application/json`?

In general, I wonder if we could be using. response headers rather than waiting for the throw.

Reply:
Here's what a diff should look like given an HTTP response from the request library:

```
HTTP RESPONSE = {
  code: 500
  statusMessage: Internal Service Error
  body: <html><title>error</html>
}
```

```diff
const [err, response] = util.makeRequest('http://...')
err instanceof ApiError === true
- err.message === 'Cannot parse response as JSON: <html><title>error</html>'
+ err.message === 'Internal Service Error'
err.body = '<html><title>error</html>'
err.response === {...raw response object...}
+ err.code === 500
```